OpenChange/MAPIStoreMailMessage.m

```diff
@@ -316,15 +316,23 @@ - (void) _fetchHeaderData
           if (i == 0)
              {
                ASSIGN (headerMimeType, mimeType);
-               ASSIGN (headerCharset, charset);               
                parameters = partParameters;
              }
-          else
+
+          if ([mimeType isEqualToString: @"text/plain"])
             {
-              /* We can only put one header charset so we will use utf-8
-                 when we have different charsets in the parts */
-              if (headerCharset != charset)
-                ASSIGN (headerCharset, @"utf-8");
+              //  utf-8 is always encoded in text/plain
+              ASSIGN (headerCharset, @"utf-8");              
```

Comment:
No, text/plain has never been encoded, and actually that was the main reason to change all of this. You send a message with `ñññññ` the body will be latin1. With this patch it will be returned as utf8, we wanted to preserve the original encoding